File: java/pattern/src/factory/simplefactory/CarFactory.java
package factory.simplefactory;

public class CarFactory {
    public static Car createCar(String type) {
        if ("audi".equals(type)) {
            return new Audi();
        } else if ("byd".equals(type)) {
            return new Byd();
        } else {
            return null;
        }
    }
}

File: php/factory/simplefactory/CarFactory.php
<?php

include 'Audi.php';
include 'Byd.php';


class CarFactory
{
    public static function createCar($type = '')
    {
        if ($type == 'audi') {
            return new Audi();
        } elseif ($type == 'byd') {
            return new Byd();
        } else {
            return null;
        }
    }
}
